scipy/stats/_discrete_distns.py

```diff
@@ -298,49 +297,47 @@ def _argcheck(self, n, p):
 
     def _pmf(self, x, n, p):
         # nbinom.pmf(k) = choose(k+n-1, n-1) * p**n * (1-p)**k
-        return exp(self._logpmf(x, n, p))
-
+        return _boost._nbinom_pdf(x, n, p)
+    
     def _logpmf(self, x, n, p):
         coeff = gamln(n+x) - gamln(x+1) - gamln(n)
         return coeff + n*log(p) + special.xlog1py(x, -p)
 
     def _cdf(self, x, n, p):
         k = floor(x)
```

Comment:
`k = floor(x)` is not used, but I it should be, otherwise we have a change in behavior.  E.g. with the master branch
```
In [6]: nbinom.cdf([1.0, 1.25, 1.5, 1.75, 2.0], 10, .4)
Out[6]: array([0.000734  , 0.000734  , 0.000734  , 0.000734  , 0.00281018])

```
and with this PR
```
In [6]: nbinom.cdf([1.0, 1.25, 1.5, 1.75, 2.0], 10, .4)                                                                                                    
Out[6]: array([0.000734  , 0.00106894, 0.00151154, 0.00208392, 0.00281018])
```
It is treating the discrete quantile as a continuous variable.

Reply:
Maybe we should dream up a test for this -- I'm surprised nothing caught it up until the human-review